ProxyManager/CacheFactory.php

```diff
@@ -1,49 +1,67 @@
 <?php
 
-namespace CacheBundle\ProxyManager;
+namespace eMAG\CacheBundle\ProxyManager;
 
-
-use CacheBundle\ProxyManager\Factory\ProxyCachingObjectFactory;
+use eMAG\CacheBundle\ProxyManager\Factory\ProxyCachingObjectFactory;
+use ProxyManager\Configuration as ProxyConfiguration;
+use ProxyManager\Version as ProxyVersion;
 
 class CacheFactory
 {
     /**
      * @var ProxyCachingObjectFactory
      */
     protected $generator;
+
     /**
-     * @var \ProxyManager\Configuration
+     * @var ProxyConfiguration
      */
     protected $proxyConfig;
 
+    /**
+     * @param   ProxyCachingObjectFactory   $generator
+     *
+     * @return  void
+     */
     public function setProxyFactory(ProxyCachingObjectFactory $generator)
     {
         $this->generator = $generator;
     }
 
-    public function setProxyConfig(\ProxyManager\Configuration $config)
+    /**
+     * @param   ProxyConfiguration  $config
+     *
+     * @return  void
+     */
+    public function setProxyConfig(ProxyConfiguration $config)
     {
         $this->proxyConfig = $config;
     }
 
+    /**
+     * @param   string  $class
+     * @param   array   $arguments
+     *
+     * @return  object
+     */
     public function generate($class, $arguments = [])
     {
         $proxyClassName = $this->proxyConfig->getClassNameInflector()->getProxyClassName($class, [
             'className' => $class,
             'factory' => ProxyCachingObjectFactory::class,
-            'proxyManagerVersion' => \ProxyManager\Version::getVersion()
+            'proxyManagerVersion' => ProxyVersion::VERSION
         ]);
```

Comment:
leave it as function. If they add something to the function... we might not generate the same names